qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py

```diff
@@ -256,6 +257,14 @@ def paulis(self, value):
             raise ValueError(
                 f"incorrect number of operators: expected {len(self.paulis)}, got {len(value)}"
             )
+        if np.any(value.phase):
```

Comment:
Since the initializer also silently maps the phase to the coefficients, I think we don't need a warning here, as this is the standard behavior. Rather, the setter was inconsistent with the initializer 🙂

Reply:
Got it, thanks for the feedback on this.

I'm still a little concerned users may get burned if they try to use the input `PauliList` for other stuff after the assignment, like
```
spo_storing_interim_result.paulis = paulis_with_phase
final_result = some_function_that_depends_critically_on_phase(paulis_with_phase)
```
but agree that 1) this PR is consistent with the initializer, 2) the above example is a somewhat unusual code structure, 3) as Jake mentioned on the issue page, this PR is kind of a band-aid since really `SparsePauliOp` should not have a phase attribute at all.

I'll go ahead and remove the warning per your suggestion.

Reply:
Andrew: I'd suggest that that code example is a problem outside `SparsePauliOp` - after the line `spo_storing_interim_result.paulis = paulis_with_phase`, it's not valid to make any assumptions about the content of `paulis_with_phase`, because the `SparsePauliOp` owns the object.  I think in general, users can hopefully either re-order the functions so that they still haven't given away ownership of the object while they're using it, or they explicitly copy it if they're still using it.

Thanks for thinking about this, though - it's good to have performance and/or mutability concerns in mind when dealing with Python.